### PR TITLE
Fix vmssflex ensureBackendPoolDeletedFromNode

### PR DESCRIPTION
**Pull request by lzhecheng**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Fix vmssflex ensureBackendPoolDeletedFromNode
* Fix loop pointer issue
* Use lock on nicUpdated var
* Fix log format
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fix vmssflex ensureBackendPoolDeletedFromNode
* Fix loop pointer issue
* Use lock on nicUpdated var
* Fix log format
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_vmss.go [hunk 1]
```diff
@@ -1906,7 +1907,9 @@ func (ss *ScaleSet) ensureBackendPoolDeleted(service *v1.Service, backendPoolIDs
 				return rerr.Error()
 			}
 
+			ss.lockMap.LockEntry(updatedVMLock)
```
Comment (jwtty) [inline pkg/provider/azure_vmss.go:1910]: Can we use atomic.Store() instead of lock?

Comment (lzhecheng) [reply pkg/provider/azure_vmss.go:1910]: For what purpose? atomic is never used.

**Pull request merged**